stratum/hal/lib/tdi/tdi_sde_helpers.cc

```diff
@@ -403,15 +403,23 @@ ::util::Status GetAllEntries(
   const ::tdi::Device *device = nullptr;
   ::tdi::DevMgr::getInstance().deviceGet(0, &device);
   const auto flags = ::tdi::Flags(0);
-  uint32 entries = 0;
+  tdi_status_t tdi_status;
+  std::unique_ptr<::tdi::Target> dev_tgt;
+  device->createTarget(&dev_tgt);
+  uint32 entries = 0, actual = 0;
+  auto table_type = static_cast<tdi_table_type_e>(table->tableInfoGet()->tableTypeGet());
+
   if (IsPreallocatedTable(*table)) {
```

Comment:
The `if` and `else` clauses are identical. You should remove the conditional and collapse the clauses into straight-line code.

I believe this also removes the need for `IsPreallocatedTable`, which was implemented to clean up a nasty piece of target-specific code. If `table->sizeGet` now works regardless of table type, you should be able to delete `IsPreallocatedTable` entirely.